src/main/java/com/aws/greengrass/lifecyclemanager/KernelLifecycle.java

```diff
@@ -180,6 +180,15 @@ public void launch() {
         logger.atInfo().setEventType("system-start").addKeyValue("main", kernel.getMain()).log();
         startupAllServices();
 
+        try {
+            GreengrassService fleetStatusService = kernel.locate(FleetStatusService.FLEET_STATUS_SERVICE_TOPICS);
+            if (fleetStatusService instanceof FleetStatusService) {
+                ((FleetStatusService) fleetStatusService).triggerFleetStatusUpdateAtKernelLaunch();
+            }
+        } catch (ServiceLoadException e) {
+            logger.atError().setCause(e).log("Failed to send status update at kernel launch because kernel was "
+                    + "unable to locate FleetStatusService");
+        }
```

Comment:
You've only requested startup for services at this point but they may not be running. Is that going to be problematic? does it make sense to wait for main to reach desired state instead?

Reply:
As we discussed offline, the nucleus launch message should be the first FSS message a device sends after launch.  If any component goes wrong while starting up, they would error and trigger their own FSS updates. 

Also FSS only needs to be initialized and postinject to send an update; doesn't have to be running.

Hence, it's preferable to not wait for main to reach desired state